packages/qunit-dom/lib/__tests__/has-tagname.ts

```diff
@@ -63,7 +63,7 @@ describe('assert.dom(...).hasTagName()', () => {
 
       expect(assert.results).toEqual([
         {
-          message: 'Element <unknown> should exist',
```

Comment:
Okay, I reverted all changes to existing tests, so the messages are the same as in `master`. To make this happen, I had to introduce a bit of screwiness to the code:

* [selectedBy](https://github.com/mainmatter/qunit-dom/pull/2087/files#diff-d38c97d68f44c96157e0203728650e0b5547fe8a550244635badd83b8b35ded3R1459-R1466) has to customize its message based on whether an element was matched or not, to mirror [the previous logic](https://github.com/mainmatter/qunit-dom/blob/a420f27f323904a1073be837f3f1909549eca095/packages/qunit-dom/lib/assertions.ts#L1224). This seems a little odd because in my mind, an argument of `null` usually means that the caller tried to pass an `Element`, but something like `querySelector()` didn't match anything and ended up evaluating to `null` and that was passing into `assert.dom()`. So I would expect to see `passed` in this case (tried to pass an `Element` that happened to be `null`), rather than `selected by null` which kinda implies a selector (string) containing `null` was passed.
* `exists()` and `isVisible()` had to do something screwy ([here](https://github.com/mainmatter/qunit-dom/pull/2087/files#diff-95990b5c77186fc52f6d6ba30f6aa46e1919752a372ea0a2202e2698935cde03R40-R42) and [here](https://github.com/mainmatter/qunit-dom/pull/2087/files#diff-1448fb716f5de96ca01505b59eed5840029a6ce50eab4852599f82c2d28a180cR41-R43)) to mirror the fact previously most assertions used `findTargetElement()` which would error with <[unknown](https://github.com/mainmatter/qunit-dom/blob/a420f27f323904a1073be837f3f1909549eca095/packages/qunit-dom/lib/assertions.ts#L1413)> if passed `null`, while `exists()` and `isVisible()` used [targetDescription](https://github.com/mainmatter/qunit-dom/blob/a420f27f323904a1073be837f3f1909549eca095/packages/qunit-dom/lib/assertions/exists.ts#L16) which called `elementToString()` which would error with <[not found](https://github.com/mainmatter/qunit-dom/blob/a420f27f323904a1073be837f3f1909549eca095/packages/qunit-dom/lib/helpers/element-to-string.ts#L4)>.

I'm fine leaving it as-is with these oddities, or creating a follow-on PR that removes them and changes some of the messaging like this PR originally did -- let me know what you prefer.